src/django/api/views/facility/facilities_view_set.py

```diff
@@ -909,7 +916,11 @@ def delete_facility_list_item(item):
                     delete_facility_match(other_match)
                     delete_facility_list_item(other_match.facility_list_item)
 
-        for claim in FacilityClaim.objects.filter(facility=facility):
+        claims = FacilityClaim.objects.filter(facility=facility)
+        for claim in claims:
+            delete_facility_claim_review_notes(claim)
```

Comment:
It is not efficient to execute two identical loops. I believe you can perform this action within the second loop.
```
        for claim in claims:
            delete_facility_claim_review_notes(claim)
            claim._change_reason = f'Deleted {facility.id}'
            claim.delete()
```